examples/ui_actions_explorer/public/app.tsx

```diff
@@ -0,0 +1,124 @@
+import { toMountPoint } from '../../../src/plugins/kibana_react/public';
+import { UiActionsStart, createAction } from '../../../src/plugins/ui_actions/public';
+import { AppMountParameters, OverlayStart } from '../../../src/core/public';
+import { HELLO_WORLD_TRIGGER_ID, HELLO_WORLD_ACTION_TYPE } from '../../ui_action_examples/public';
+import { TriggerContextExample } from './trigger_context_example';
+
+interface Props {
+  uiActionsApi: UiActionsStart;
+  openModal: OverlayStart['openModal'];
+}
+
+const ActionsExplorer = ({ uiActionsApi, openModal }: Props) => {
+  const [name, setName] = useState('Waldo');
+  const [confirmationText, setConfirmationText] = useState('');
+  return (
+    <EuiPage>
+      <EuiPageBody>
+        <EuiPageHeader>Ui Actions Explorer</EuiPageHeader>
+        <EuiPageContent>
+          <EuiPageContentBody>
+            <EuiText>
+              <p>
+                By default there is a single action attached to the `HELLO_WORLD_TRIGGER`. Clicking
+                this button will cause it to be executed immediately.
+              </p>
+            </EuiText>
+            <EuiButton
+              data-test-subj="emitHelloWorldTrigger"
+              onClick={() => uiActionsApi.executeTriggerActions(HELLO_WORLD_TRIGGER_ID, {})}
+            >
+              Say hello world!
+            </EuiButton>
+
+            <EuiText>
+              <p>
+                Lets dynamically add new actions to this trigger. After you click this button, click
+                the above button again. This time it should offer you multiple options to choose
+                from. Using the UI Action and Trigger API makes your plugin extensible by other
+                plugins. Any actions attached to the `HELLO_WORLD_TRIGGER_ID` will show up here!
+              </p>
+              <EuiFieldText prepend="Name" value={name} onChange={e => setName(e.target.value)} />
+              <EuiButton
+                data-test-subj="addDynamicAction"
+                onClick={() => {
+                  const dynamicAction = createAction<{}>({
+                    type: `${HELLO_WORLD_ACTION_TYPE}-${name}`,
+                    getDisplayName: () => `Say hello to ${name}`,
+                    execute: async () => {
+                      const overlay = openModal(
+                        toMountPoint(
+                          <EuiModalBody>
+                            <EuiText data-test-subj="dynamicHelloWorldActionText">
+                              {`Hello ${name}`}
+                            </EuiText>{' '}
+                            <EuiButton data-test-subj="closeModal" onClick={() => overlay.close()}>
+                              Close
+                            </EuiButton>
+                          </EuiModalBody>
+                        )
+                      );
+                    },
+                  });
+                  uiActionsApi.registerAction(dynamicAction);
```

Comment:
should we be cleaning this up somewhere ?

Reply:
No, I don't think so, though a page refresh will remove the dynamically added actions.  I added a note in the success callout: 

`Refresh the page to reset state.  It's up to the user of the system to persist state like this.`